pallets/polkapool/src/lib.rs

```diff
@@ -19,131 +19,160 @@
+    + Dispatchable<Origin=<Self as Config>::Origin>
+    + GetDispatchInfo
+    + From<frame_system::Call<Self>>;
     /// Balance Type
     type Balance: Parameter
-        + Member
-        + AtLeast32BitUnsigned
-        + Default
-        + Copy
-        + MaybeSerializeDeserialize
-        + Clone
-        + Zero
-        + One
-        + PartialOrd
-        + Bounded;
+    + Member
+    + AtLeast32BitUnsigned
+    + Default
+    + Copy
+    + MaybeSerializeDeserialize
+    + Clone
+    + Zero
+    + One
+    + PartialOrd
+    + Bounded;
     /// Module that handles tokens
     type Currency: MultiReservableCurrency<
         Self::AccountId,
-        CurrencyId = AssetId,
-        Balance = Self::Balance,
-    >;
+        CurrencyId=AssetId,
+        Balance=Self::Balance,
+    > + MultiLockableCurrency<Self::AccountId>;
     /// Min amount that must be staked
     type MinStakeAmount: Get<Self::Balance>;
     /// Maximum allowed Feeless Transactions in a block
     type MaxAllowedWeight: Get<Weight>;
     /// Min Stake Period
-    type MinStakePeriodPerWeight: Get<u32>;
+    type MinStakePeriod: Get<BlockNumber>;
+    /// Max number of stakes per account
+    type MaxStakes: Get<usize>;
     /// Randomness Source
     type RandomnessSource: Randomness<H256, BlockNumber>;
     /// Call Filter
     type CallFilter: Filter<<Self as Config>::Call>;
-    //Minimum Stake per Call
+    /// Minimum Stake per Call
     type MinStakePerWeight: Get<u128>;
-
-    type GovernanceOrigin: EnsureOrigin<<Self as Config>::Origin, Success = Self::AccountId>;
+    /// The Governance Origin that can slash stakes
+    type GovernanceOrigin: EnsureOrigin<<Self as Config>::Origin, Success=Self::AccountId>;
 }
 
 #[derive(Decode, Encode, Copy, Clone)]
-pub struct StakeInfo<T: Config + frame_system::Config> {
+pub struct Stake<T: Config + frame_system::Config> {
     pub staked_amount: T::Balance,
     pub unlocking_block: T::BlockNumber,
 }
 
-impl<T: Config + frame_system::Config> Default for StakeInfo<T> {
+impl<T: Config + frame_system::Config> Default for Stake<T> {
     fn default() -> Self {
-        StakeInfo {
+        Stake {
             staked_amount: 0_u128.saturated_into(),
             unlocking_block: 1_u32.saturated_into(),
         }
     }
 }
 
-impl<T: Config + frame_system::Config> StakeInfo<T> {
-    pub fn new(stake: T::Balance, unlock: T::BlockNumber) -> StakeInfo<T> {
-        StakeInfo {
+impl<T: Config + frame_system::Config> Stake<T> {
+    pub fn new(stake: T::Balance, unlock: T::BlockNumber) -> Stake<T> {
+        Stake {
             staked_amount: stake,
             unlocking_block: unlock,
         }
     }
 }
 
-#[derive(Decode, Encode, Copy, Clone)]
-pub struct MovingAverage<T: Config> {
-    pub amount: <T as Config>::Balance,
-    pub count: <T as Config>::Balance,
+#[derive(Decode, Encode, Clone)]
+pub struct StakeInfo<T: Config + frame_system::Config> {
+    stakes: VecDeque<Stake<T>>,
```

Comment:
Defined a vecdeque for storing the different locks aka stakes for different extrinsics